### ISSUE TITLE
Getting "Invalid API key" error, but the key is valid

### ISSUE THREAD

**Issue by vyorkin** [Lucsanszky/haskell-bitmex#14]
Hey @Lucsanszky, thanks for your work on this package!
Unfortunately, I'm unable to run the provided example, could you please take a quick look? Maybe I'm just missing smth, not sure what I'm doing wrong...

My bitmex testnet keys:
key: `8hdxgs6ZXaaiUFH9lCrU9R9Y`
secret: `WLidt39_GZUPjY1RTLnzaKWjNt1tn_zONWXt_zu9HSTATFdW`

![image](https://user-imageS.githubusercontent.com/988849/48205448-4d2cd100-e37d-11e8-8cb7-644c25e4788d.png)

stdout:

```sh
Just (Info (INFO {info = "Welcome to the BitMEX Realtime API.", version = "2018-11-05T18:47:31.000Z", timestamp = "2018-11-08T14:33:09.591Z", docs = "https://testnet.bitmex.com/app/wsAPI", limit = Object (fromList [("remaining",Number 39.0)])}))
MimeResult {mimeResult = Left (MimeError {mimeError = "error statusCode: 401", mimeErrorResponse = Response {responseStatus = Status {statusCode = 401, statusMessage = "Unauthorized"}, responseVersion = HTTP/1.1, responseHeaders = [("Date","Thu, 08 Nov 2018 14:33:12 GMT"),("Content-Type","application/json; charset=utf-8"),("Content-Length","98"),("Connection","keep-alive"),("ETag","W/\"62-o96CYdSglPbBUdDLXP2ovufS25s\""),("Strict-Transport-Security","max-age=31536000; includeSubDomains")], responseBody = "{\"error\":{\"message\":\"Missing authorization headers: api-nonce or api-expires\",\"name\":\"HTTPError\"}}", responseCookieJar = CJ {expose = []}, responseClose' = ResponseClose}}), mimeResultResponse = Response {responseStatus = Status {statusCode = 401, statusMessage = "Unauthorized"}, responseVersion = HTTP/1.1, responseHeaders = [("Date","Thu, 08 Nov 2018 14:33:12 GMT"),("Content-Type","application/json; charset=utf-8"),("Content-Length","98"),("Connection","keep-alive"),("ETag","W/\"62-o96CYdSglPbBUdDLXP2ovufS25s\""),("Strict-Transport-Security","max-age=31536000; includeSubDomains")], responseBody = "{\"error\":{\"message\":\"Missing authorization headers: api-nonce or api-expires\",\"name\":\"HTTPError\"}}", responseCookieJar = CJ {expose = []}, responseClose' = ResponseClose}}
Just (Info (INFO {info = "Welcome to the BitMEX Realtime API.", version = "2018-11-05T18:47:31.000Z", timestamp = "2018-11-08T14:33:10.231Z", docs = "https://testnet.bitmex.com/app/wsAPI", limit = Object (fromList [("remaining",Number 38.0)])}))
Just (Error (ERROR {error = "Invalid API Key.", status = Just 401, meta = Just (Object (fromList [])), request = Just (Object (fromList [("op",String "authKey"),("args",Array [String "8hdxgs6ZXaaiUFH9lCrU9R9Y\n",Number 1.541687590274643e15,String "a154bd8567f257de97680e6469d724ee15fef574cea4ebe99a3c443132451c78"])]))}))
example: CloseRequest 1000 ""
```

**Comment by Lucsanszky** [comment]
Hey @vyorkin, thanks for reporting this! I think I found the problem: I suspect that you might have a carriage return in your API key files. 

Here's what I did:
```
echo "8hdxgs6ZXaaiUFH9lCrU9R9Y" >> ~/.ssh/vyorkin-bitmex.pub
echo "WLidt39_GZUPjY1RTLnzaKWjNt1tn_zONWXt_zu9HSTATFdW" >> ~/.ssh/vyorkin-bitmex
cabal new-run example ~/.ssh/vyorkin-bitmex.pub ~/.ssh/vyorkin-bitmex
```
This reproduced the error you encountered. After that, I noticed the newline character at the end of the files, I proceeded to remove them and then everything worked as expected.
Try this and let me know if it worked for you.

Also, I'll probably update the API and filter out whitespaces in the API key files, so this inconvenience won't happen again in the future. :)
